src/uu/cp/src/cp.rs

```diff
@@ -2475,12 +2471,10 @@ fn copy_fifo(dest: &Path, overwrite: OverwriteMode, debug: bool) -> CopyResult<(
         fs::remove_file(dest)?;
     }
 
-    let name = CString::new(dest.as_os_str().as_bytes()).unwrap();
-    let err = unsafe { mkfifo(name.as_ptr(), 0o666) };
-    if err == -1 {
-        return Err(format!("cannot create fifo {}: File exists", dest.quote()).into());
+    match make_fifo(dest) {
+        Ok(_) => Ok(()),
+        Err(_) => Err(format!("cannot create fifo {}: File exists", dest.quote()).into()),
     }
```

Comment:
```suggestion
makefifo(dest).map_err(|_| format!("cannot create fifo {}: File exists", dest.quote()).into())
```

probably more rust idiomatic